File: Codeforces/918/E_Romantic_Glasses.py
import sys
input = sys.stdin.readline

def solve():
    n = int(input())
    ar = list(map(int,input().split()))
    mp = {0:1}
    sum=0
    for i in range(n):
        if(i&1): ar[i]*=-1
        sum+=ar[i]
        if sum in mp:
            print("YES")
            return
        mp[sum]=1
    
    print("NO")

t = int(input())
for _ in range(t): solve()

File: Codeforces/918/E_Romantic_Glasses.cpp
/*----------------------------------------
    author:  Imam
    created: 09.07.2025 17:17:13
----------------------------------------*/
#include<bits/stdc++.h>
using namespace std;
inline void solve()
{
    int n; cin>>n;
    vector<int64_t> ar(n);
    for(auto &i:ar) cin>>i;
    map<int64_t,int64_t> mp;
    int64_t sum=0;
    mp[sum]++;
    for(int i=0;i<n;i++){
        if(i%2)ar[i]*=-1;
        sum+=ar[i];
        if(mp[sum]){cout<<"YES\n";return;}
        mp[sum]++;
    }
    cout<<"NO\n";
}
int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr); cout.tie(NULL);
    
    int t=1; cin>>t;
    while(t--) solve();
    return 0;
}